src/subsystems/haskell/translators/stack-lock/utils.nix

```diff
@@ -36,39 +36,4 @@ in rec {
     '';
   in
     l.fromJSON (l.readFile jsonFile);
-
-  # converts all cabal files for a given list of candidates to json files
-  batchCabal2Json = candidates: let
-    candidatesJsonStr = l.toJSON candidates;
-    convertOne = name: version: ''
-      cabalFile=${all-cabal-hashes}/${name}/${version}/${name}.cabal
-      if [ -e $cabalFile ]; then
-        echo "converting cabal to json: ${name}-${version}"
-        mkdir -p $out/${name}/${version}
-        ${cabal2json}/bin/cabal2json \
-          $cabalFile \
-          > $out/${name}/${version}/cabal.json
-      else
-        echo "all-cabal-hashes" seems to be outdated
-        exit 1
```

Comment:
I forgot about including something like this in my modified versions of this function. Is there a clean way of doing this check without Bash?

Reply:
OK, I used `throw`.